apps/docs/content/guides/ai/hybrid-search.mdx

```diff
@@ -0,0 +1,228 @@
+- `content` contains the actual text we will be searching over.
+- `fts` is an auto-generated `tsvector` column that is generated using the text in `content`. We will use this for [full text search](/docs/guides/database/full-text-search) (search by keyword).
+- `embedding` is a [vector column](/docs/guides/ai/vector-columns) that stores the vector generated from our embedding model. We will use this for [semantic search](/docs/guides/ai/concepts#what-are-embeddings) (search by meaning). We chose 512 dimensions for this example, but adjust this to match the size of the embedding vectors generated from your preferred model.
+
+Next we'll create indexes on the `fts` and `embedding` columns so that their individual queries will remain fast at scale:
+
+```sql
+-- Create an index for the full-text search
+create index on documents using gin(fts);
+
+-- Create an index for the semantic vector search
+create index on documents using hnsw (embedding vector_ip_ops);
+```
+
+For full text search we use a [generalized inverted (GIN) index](https://www.postgresql.org/docs/current/gin-intro.html) which is designed for handling composite values like those stored in a `tsvector`.
+
+For semantic vector search we use an [HNSW index](/docs/guides/ai/vector-indexes/hnsw-indexes), which is a high performing approximate nearest neighbor (ANN) search algorithm. Note that we are using the `vector_ip_ops` (inner product) operator with this index because we plan on using the inner product (`<#>`) operator later in our query. If you plan to use a different operator like cosine distance (`<=>`), be sure to update the index accordingly. For more information, see [distance operators](/docs/guides/ai/vector-indexes#distance-operators).
+
+Finally we'll create our `hybrid_search` function:
+
+```sql
+create or replace function hybrid_search(
+  query_text text,
+  query_embedding vector(512),
+  match_count int,
+  full_text_weight float = 1,
+  semantic_weight float = 1,
+  rrf_k int = 50
+)
+returns setof documents
+language sql
+as $$
+with full_text as (
+  select
+    id,
+    row_number() over() as rank_ix
+  from
+    documents
+  where
+    fts @@ websearch_to_tsquery(query_text)
+  order by
+    -- Note: this is not indexable but will only rank matches of the where clause
+    -- which shouldn't be too big
+    ts_rank_cd(fts, websearch_to_tsquery(query_text)) desc
+  limit match_count * 2
+),
+semantic as (
+  select
+    id,
+    row_number() over () as rank_ix
+  from
+    documents
+  order by
+    doc_vector <#> query_embedding
+  limit match_count * 2
+)
+select
+  documents.*
+from
+  full_text
+  full outer join semantic
+    on full_text.id = semantic.id
+  join documents
+    on coalesce(full_text.id, semantic.id) = documents.id
+order by
+  coalesce(1.0 / (rrf_k + full_text.rank_ix), 0.0) * full_text_weight +
+  coalesce(1.0 / (rrf_k + semantic.rank_ix), 0.0) * semantic_weight
+  desc
```

Comment:
I added optional weight arguments to the function (defaults to 1 for each). They allow you to modify how much weight semantic search gets vs fts.